assets/cluster-monitoring-operator/prometheus-rule.yaml

```diff
@@ -30,6 +30,16 @@ spec:
       for: 15m
       labels:
         severity: warning
+    - alert: PrometheusPossibleNarrowSelectors
+      annotations:
+        description: Queries or/and relabel configs on Prometheus/Thanos {{$labels.namespace}}/{{$labels.pod}} could be too restrictive.
+        summary: |
+          Some queries or/and relabel configs with selectors on the values of the "le" label of classic histograms or/and the "quantile" label of summaries
+          may not take into account that values could also be floats, they may need to be adjusted. If assistance is needed, please let us know at https://issues.redhat.com/browse/MON-4129.
+      expr: increase(prometheus_narrow_selectors_count{job=~"prometheus-k8s|prometheus-user-workload|thanos-querier|thanos-ruler"}[5m]) > 0
```

Comment:
is metrics `prometheus_narrow_selectors_count` since prometheus 3?

Reply:
Yes, It'll be added downstream on top of v3 https://github.com/openshift/prometheus/pull/227/commits/3e1c833e3c96149a63fa1148c96d30ef9357ab8e and on thanos https://github.com/openshift/thanos/pull/158

We'll decide later if we want to release it, but in the meantime, it should continue helping to identify misuses during dev.